app/lib/three_scale/search.rb

```diff
@@ -186,7 +186,8 @@ module Helpers
     MAX_PER_PAGE = 20
 
     # By default sphinx paginates search results, and the default page size is 20.
-    # Setting to a high value "disables" sphinx-based pagination to use will_paginate instead
+    # Setting to a high value "disables" sphinx-based pagination when full results are needed.
+    # e.g. Searchable::by_query scope which would not work properly otherwise.
```

Comment:
```suggestion
    # There’s no way to turn it off, but we can request really big pages as a workaround.
    # e.g Searchable::by_query implements this workaround.
```

Reply:
What is `by_query` for anyway? Where is it used? Maybe we should add this comment there?

I think it's a bit confusing to comment on a constant. Maybe we should rename it in order to make self-explanatory...? It's terrible, I know.
```diff
-SPHINX_PAGE_SIZE_INFINITE = 1_000_000
+PAGE_SIZE_INFINITE_TO_PREVENT_SEARCH_PAGINATION = 1_000_000
```

Reply:
Uf, I'm sorry, but I'd prefer a comment rather than such a long constant name :zany_face: 

Otherwise, we could have something like `PAGE_SIZE_INFINITE_TO_PREVENT_SEARCH_PAGINATION_BY_SPHINX_AND_PAGINATE_WITH_WILL_PAGINATE`

Reply:
>What is by_query for anyway? Where is it used? Maybe we should add this comment there?
Yeah, maybe we could expand further that we search in two steps:
1. find all IDs in sphinx that match the keyword
2. join `where(id: found_ids)` with other scopes to get the final results

And apply pagination over the results of 2, so in 1. we need to get all IDs, disabling sphinx-provided pagination.

Reply:
This workaround has nothing to do with `will_paginate`. It is needed whether `will_paginate` is being used or not. I like @josemigallas suggestion but want to make sure I'm not missing something regarding `will_paginate`, so please explain to me if I'm wrong.

wrt constant name - I also don't like such long names. And still comment provides more detailed and useful information. So not a significant improvement to justify the change.

Reply:
OK, yeah, `will_paginate` is not quite relevant indeed. We just need to skip pagination done by sphinx (and don't care at this point whether the final results will be paginated or not at all).